### ISSUE TITLE
ChangeTracker don't see changes in navigation property collection

### ISSUE THREAD

**Issue by lambidu** [dotnet/efcore#12335]
I'm trying to get all changes made to an entity, including any changes made on navigation properties which belongs to that entity.

So, I've made an extension which recursively walks trough entity properties and all properties on it's references, and collects a list of what changed.

Here is my extension:
```c#
public static IEnumerable<PropertyEntry> GetChanges(this EntityEntry entity) {
	// Get changed properties
	var changes = entity.Properties
		.Where(m => m.IsModified == true && m.IsTemporary == false)
		.ToList();

	// Get changes from any navigation property
	var references = entity.References
		.Select(m => m.TargetEntry)
		.ToList();

	// Iterate navigation properties changes
	foreach (var reference in references) {
		// Add navigation properties changes to changes list
		changes.AddRange(reference.GetChanges());
	}

	// Return complete list of detected changes
	return changes;
}
```
and I call it from my repository class like so:
```c#
// ...
this.context.Entry(entity).GetChanges();
// ...
```
Entity class (User, inherited from IdentityUser), has about 8 navigation collections (`Claims`, `Roles`, `AuthorizedIPs`, ...).

Now, when I'm trying change `AuthorizedIPs` collection (add one new entry), I see `CollectionEntry.CurrentValue` containing newly added entry, but `CollectionEntry.IsModified` is still `false`

![image](https://user-images.githubusercontent.com/2502621/41284684-a66f8d44-6e42-11e8-9a82-718620855e02.png)

### Further technical details
EF Core version: 2.0.3
Database Provider: Pomelo.EntityFrameworkCore.MySql 2.0.1
Operating system: Windows 10
IDE: (e.g. Visual Studio 2017 15.7.3)

**Comment by ajcvickers** [comment]
Duplicate of #10450

**Comment by crackyTozo** [comment]
Was this issue ever solved or was there a solution provided? I'm experiencing similar problem in which a property change or update is not persisted to the database because the property is part of a navigation property key. Hence, it's not mark as "modified". I can't find a solution.

**Comment by ajcvickers** [comment]
@crackyTozo The `closed-duplicate` indicates that this issue was closed as a duplicate of #10450
